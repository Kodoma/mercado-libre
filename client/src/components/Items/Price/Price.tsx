import React from 'react';
import styles from '../Items.module.css';
import { Link, Typography } from '@material-ui/core';
import Image from 'next/image';
import NumberFormat from 'react-number-format';
import { useRouter } from 'next/router';

const Price: React.FunctionComponent<PriceProps> = ({ id, amount, currency, free_shipping }) => {
    const router = useRouter();
    const { search } = router.query;
    return (
        <div className={styles.priceContainer}>
            {search ? (
                <Link href={`/items/${id}`} className={styles.price}>
                    <Typography variant="h6">
                        <NumberFormat
                            value={amount}
                            displayType={'text'}
                            thousandSeparator={true}
                            prefix={currency}
                        />
                    </Typography>
                </Link>
            ) : (
                <Typography variant="h4" className={styles.price}>
                    <NumberFormat
                        value={amount}
                        displayType={'text'}
                        thousandSeparator={true}
                        prefix={currency}
                    />
                </Typography>
            )}
            {free_shipping && (
                <div className={styles.freeShipping}>
                    <Image
                        src={
                            search
                                ? '/images/ic_shipping_list.png'
                                : '/images/ic_shipping_vip.png'
                        }
                        alt="Envios Gratis"
                        width={search ? '18px' : '32px'}
                        height={search ? '18px' : '32px'}
                    />
                </div>
            )}
        </div>
    );
};

export default React.memo(Price);

interface PriceProps {
    id: string;
    amount: string;
    currency: string;
    free_shipping: boolean;
}

interface PriceProps {
    id: string;
    amount: string;
    currency: string;
    free_shipping: boolean;
}
