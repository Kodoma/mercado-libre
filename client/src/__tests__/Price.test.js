import React from 'react';
import 'jsdom-global/register';
import {mount, shallow} from 'enzyme';
import Price from "../components/Items/Price/Price";

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('__tests__ Price', () => {

    it('should render my Price component', () => {
        useRouter.mockImplementationOnce(() => ({
            query: { search: 'iphone' },
        }))

        const component = shallow(<Price
            id={"1"}
            amount={"100"}
            currency={"ARS"}
            free_shipping={false}
        />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('should render Price and check value and currency', function () {
        useRouter.mockImplementationOnce(() => ({
            query: { search: 'iphone' },
        }))

        const wrap = mount(<Price
                                id={"1"}
                                amount={"100"}
                                currency={"ARS"}
                                free_shipping={false}
                            />)
        expect(wrap.find('div').text()).toBe('ARS100')
    })
});
