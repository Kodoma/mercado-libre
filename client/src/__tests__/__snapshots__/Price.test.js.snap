// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__tests__ Price should render my Price component 1`] = `
Array [
  <div
    className="priceContainer"
  >
    <WithStyles(ForwardRef(Link))
      className="price"
      href="/items/1"
    >
      <WithStyles(ForwardRef(Typography))
        variant="h6"
      >
        <NumberFormat
          allowEmptyFormatting={false}
          allowLeadingZeros={false}
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          fixedDecimalScale={false}
          isAllowed={[Function]}
          isNumericString={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          onValueChange={[Function]}
          prefix="ARS"
          suffix=""
          thousandSeparator={true}
          thousandsGroupStyle="thousand"
          type="text"
          value="100"
        />
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Link))>
  </div>,
]
`;
